File: sketch/src/main/java/me/xiaopan/sketch/ImageFormat.java
package me.xiaopan.sketch;

import android.graphics.Bitmap;

public enum ImageFormat {
    PNG("image/png", Bitmap.Config.ARGB_8888, Bitmap.Config.ARGB_4444),
    JPEG("image/jpeg", Bitmap.Config.RGB_565, Bitmap.Config.RGB_565),
    GIF("image/gif", Bitmap.Config.ARGB_8888, Bitmap.Config.ARGB_4444),
    BMP("image/bmp", Bitmap.Config.RGB_565, Bitmap.Config.RGB_565),
    WEBP("image/webp", Bitmap.Config.ARGB_8888, Bitmap.Config.ARGB_4444);

    String mimeType;
    Bitmap.Config bestConfig;
    Bitmap.Config lowQualityConfig;

    ImageFormat(String mimeType, Bitmap.Config bestConfig, Bitmap.Config lowQualityConfig) {
        this.mimeType = mimeType;
        this.bestConfig = bestConfig;
        this.lowQualityConfig = lowQualityConfig;
    }

    public Bitmap.Config getConfig(boolean lowQualityImage) {
        return lowQualityImage?lowQualityConfig:bestConfig;
    }

    public String getMimeType() {
        return mimeType;
    }

    public void setBestConfig(Bitmap.Config bestConfig) {
        this.bestConfig = bestConfig;
    }

    public void setLowQualityConfig(Bitmap.Config lowQualityConfig) {
        this.lowQualityConfig = lowQualityConfig;
    }

    public boolean equals(String mimeType){
        return this.mimeType.equalsIgnoreCase(mimeType);
    }

    public static ImageFormat valueOfMimeType(String mimeType){
        for(ImageFormat imageFormat : values()){
            if(imageFormat.equals(mimeType)){
                return imageFormat;
            }
        }
        return null;
    }
}

File: sketch/src/main/java/com/github/panpf/sketch/ImageFormat.kt
package com.github.panpf.sketch

enum class ImageFormat(val mimeType: String) {
    JPEG("image/jpeg"),
    PNG("image/png"),
    WEBP("image/webp"),
    GIF("image/gif"),
    BMP("image/bmp"),
    HEIC("image/heic"),
    HEIF("image/heif"),
    ;

    companion object {
        @JvmStatic
        fun valueOfMimeType(mimeType: String?): ImageFormat? = when {
            JPEG.mimeType.equals(mimeType, ignoreCase = true) -> JPEG
            PNG.mimeType.equals(mimeType, ignoreCase = true) -> PNG
            WEBP.mimeType.equals(mimeType, ignoreCase = true) -> WEBP
            GIF.mimeType.equals(mimeType, ignoreCase = true) -> GIF
            BMP.mimeType.equals(mimeType, ignoreCase = true) -> BMP
            HEIC.mimeType.equals(mimeType, ignoreCase = true) -> HEIC
            HEIF.mimeType.equals(mimeType, ignoreCase = true) -> HEIF
            else -> null
        }
    }
}